### PR TITLE
Fix static analysis errors relevant to video codecs

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: src/h264/h264encoder.cpp [hunk 1]
```diff
@@ -68,6 +68,15 @@ H264Encoder::H264Encoder(const Properties& properties) : frame(VideoCodec::H264)
 	type    = VideoCodec::H264;
 	format  = 0;
 	pts     = 0;
+	curNal = 0;
```
Comment (murillo128) [inline src/h264/h264encoder.cpp:71]: can we move it to the .h instead?

File: src/vp8/vp8encoder.cpp [hunk 2]
```diff
@@ -31,7 +31,16 @@ VP8Encoder::VP8Encoder(const Properties& properties) : frame(VideoCodec::VP8)
 	type    = VideoCodec::VP8;
 	pts	= 0;
 	num	= 0;
-	pic	= NULL;
+	pic	= NULL;
```
Comment (murillo128) [inline src/vp8/vp8encoder.cpp:34]: can we move it to the .h instead?

File: src/h264/h264encoder.cpp [hunk 3]
```diff
@@ -68,6 +68,15 @@ H264Encoder::H264Encoder(const Properties& properties) : frame(VideoCodec::H264)
...
+	numNals = 0;
+	width = 0;
+	height = 0;
+	nals = 0;
+	
+	memset(&pic, 0, sizeof(pic));
```
Comment (murillo128) [inline src/h264/h264encoder.cpp:77]: `pic = {}` in .h instead?

File: include/VideoDecoderWorker.h [hunk 4]
```diff
@@ -32,7 +32,7 @@ class VideoDecoderWorker
 private:
 	std::set<VideoOutput*> outputs;
 	WaitQueue<RTPPacket::shared> packets;
-	pthread_t thread;
+	pthread_t thread = 0;
```
Comment (murillo128) [inline include/VideoDecoderWorker.h:35]: nullptr or {} better?

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: src/vp8/vp8encoder.cpp (hunk 2 above)
Comment (harryz2000) [reply src/vp8/vp8encoder.cpp:34]: I tried to make small change and keep the initialisation consistent with existing code. Would you like to change the existing code as well? Sometimes the change would be big and probably not be convenient for review.

File: include/VideoDecoderWorker.h (hunk 4 above)
Comment (harryz2000) [reply include/VideoDecoderWorker.h:35]: As mentioned in another pr. pthread_t is an integer, seems 0 is more suitable

File: src/h264/h264encoder.cpp (hunk 3 above)
Comment (harryz2000) [reply src/h264/h264encoder.cpp:77]: Yes. We can. As mentioned in another response, how should we resolve this as existing code does initialisation in cpp file. Which way should we take?
1. moving all initialisation to header file, including existing code
2. Only add initialisation for missing ones, don't change existing ones
3. Keep the missing ones consistent with existing code
I am fine to either way.

Comment (murillo128) [reply src/h264/h264encoder.cpp:77]: The code may be bigger, but the complexity is low. So let's modernize it and go with 3.

Comment (harryz2000) [reply src/h264/h264encoder.cpp:77]: I assume you meant: Make all initialisation consistent and in header file. I will change to that way. Please let me know if my understanding is correct.

Comment (murillo128) [reply src/h264/h264encoder.cpp:77]: that's correct

File: src/h264/h264encoder.cpp (hunk 1 above)
Comment (harryz2000) [reply src/h264/h264encoder.cpp]: Done

File: src/vp8/vp8encoder.cpp (hunk 2 above)
Comment (harryz2000) [reply src/vp8/vp8encoder.cpp]: Done

File: src/h264/h264encoder.cpp (hunk 3 above)
Comment (harryz2000) [reply src/h264/h264encoder.cpp]: Done.

Comment (harryz2000) [reply src/h264/h264encoder.cpp]: Actually let me double check whether {} has same effect as memset. The {} does eliminate the Coverity errors.

Comment (harryz2000) [reply src/h264/h264encoder.cpp]: Yes. My tests indicated it does zero the struct.

**Review by murillo128 — APPROVED** [review]
lgtm

**Pull request merged**